mapbox-lint/src/main/java/com/mapbox/maps/lint/LifecycleMethodDetector.kt

```diff
@@ -0,0 +1,47 @@
+package com.mapbox.maps.lint
+
+import com.android.tools.lint.detector.api.*
+import com.intellij.psi.PsiMethod
+import org.jetbrains.uast.UCallExpression
+
+/**
+ * Lint detector for lifecycle events
+ */
+class LifecycleMethodDetector : Detector(), Detector.UastScanner {
+
+  override fun getApplicableMethodNames() = listOf(ON_START, ON_STOP, ON_DESTROY, ON_LOW_MEMORY)
+
+  override fun visitMethodCall(context: JavaContext, node: UCallExpression, method: PsiMethod) {
+    if (context.evaluator.isMemberInClass(method, "com.mapbox.maps.MapView")) {
+      context.report(
+        ISSUE, node, context.getLocation(node),
+        REPORT_MESSAGE
+      )
+    }
+  }
+
+  companion object Issues {
+    private const val ON_START = "onStart"
+    private const val ON_STOP = "onStop"
+    private const val ON_DESTROY = "onDestroy"
+    private const val ON_LOW_MEMORY = "onLowMemory"
+    private const val ISSUE_ID = "Lifecycle"
+    private const val BRIEF_DESCRIPTION = "Redundant method call"
+    private const val PRIORITY = 8
+    private const val REPORT_MESSAGE =
+      "No need to invoke onStart/onStop/onDestroy/onLowMemory explicitly when Mapbox Lifecycle Plugin is added as dependency."
```

Comment:
we should also mention the limitation of appcompact 1.3.0 as well here. Since it would be required to overwrite manually if the user uses appcompact before 1.3.0

Reply:
```suggestion
      "No need to invoke onStart/onStop/onDestroy/onLowMemory explicitly when Mapbox Lifecycle Plugin is added as dependency and appcompact 1.3.0+ is used."
```